sns/src/test/scala/akka/stream/alpakka/sns/SnsPublishSinkSpec.scala

```diff
@@ -0,0 +1,125 @@
+package akka.stream.alpakka.sns
+
+import java.util.concurrent.{CompletableFuture, Future}
+
+import akka.Done
+import akka.stream.alpakka.sns.scaladsl.SnsPublishSink
+import akka.stream.scaladsl.Keep
+import akka.stream.testkit.scaladsl.TestSource
+import com.amazonaws.handlers.AsyncHandler
+import com.amazonaws.services.sns.{AmazonSNSAsync, AmazonSNSAsyncClient}
+import com.amazonaws.services.sns.model.{PublishRequest, PublishResult}
+import org.mockito.ArgumentMatchers.{any, eq => meq}
+import org.mockito.Mockito._
+import org.mockito.invocation.InvocationOnMock
+import org.mockito.stubbing.Answer
+import org.scalatest.{FlatSpec, MustMatchers}
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class SnsPublishSinkSpec extends FlatSpec with DefaultTestContext with MustMatchers {
+
+  it should "publish a single message to sns" in {
+    implicit val snsClient: AmazonSNSAsync = mock[AmazonSNSAsyncClient]
+    when(snsClient.publishAsync(any[PublishRequest](), any())).thenAnswer(
+      new Answer[AnyRef] {
+        override def answer(invocation: InvocationOnMock): Future[PublishResult] = {
+          val publishRequest = invocation.getArgument[PublishRequest](0)
+          invocation
+            .getArgument[AsyncHandler[PublishRequest, PublishResult]](1)
+            .onSuccess(
+              publishRequest,
+              new PublishResult().withMessageId("message-id")
+            )
+          new CompletableFuture()
```

Comment:
This future is never completed though, shouldn't that cause the `matVal` to never complete, is there a bug around waiting for ack on publishing?